### ISSUE TITLE
takePicture failed / autoFocus failed

### ISSUE THREAD

**Issue by philipgiuliani** [boxme/SquareCamera#26]
Hi, im looking through my crashlogs of the app today, and some users are experiencing weird errors:
```java
java.lang.RuntimeException: takePicture failed
       at android.hardware.Camera.native_takePicture(Camera.java)
       at android.hardware.Camera.takePicture(Camera.java:1484)
       at com.keepinmind.lifeshot.ui.CameraFragment.takePicture()
       at com.keepinmind.lifeshot.ui.CameraFragment.access$900()
       at com.keepinmind.lifeshot.ui.CameraFragment$4.onClick()
```
See: http://stackoverflow.com/questions/21723557/java-lang-runtimeexception-takepicture-failed

And the following:
```java
java.lang.RuntimeException: autoFocus failed
       at android.hardware.Camera.native_autoFocus(Camera.java)
       at android.hardware.Camera.autoFocus(Camera.java:1299)
       at com.keepinmind.lifeshot.widget.SquareCameraPreview.handleFocus()
       at com.keepinmind.lifeshot.widget.SquareCameraPreview.onTouchEvent()
```
See: http://stackoverflow.com/questions/8820702/autofocus-throwing-exception
See: https://github.com/dm77/barcodescanner/commit/15a64d00dfb95dff8e7a9c0ef87a1ad474a220fd

**Comment by wdtaylor** [comment]
To reproduce this error.  Do the following.
Launch camera
Trigger phone screen lock
Unlock screen.
Notice still preview
Tap screen or take picture button to cause crash.

**Comment by GiacomoDB** [comment]
@wdtaylor 
You can easily solve that issue working on the **OnResume**:

     @Override
    public void onResume() 
    {
        // stop the preview
        if (mCamera != null) 
        {
            //we are good 
        }
        else
        {
        	restartPreview();
        }
        super.onResume();
    }

**Comment by boxme** [comment]
Resolved in a511afcd1aba34a235074ba4cc957675c66b19f9 and 97c88eb9c240b69cfde3737aeea6e93216c65888

**Comment by fajaranugrah** [comment]
any solution i got error on same this

android.hardware.Camera.native_takePicture (Camera.java)
android.hardware.Camera.takePicture (Camera.java:1570)
com.desmond.squarecamera.CameraFragment.takePicture (CameraFragment.java:425)
com.desmond.squarecamera.CameraFragment.access$1000 (CameraFragment.java:33)
com.desmond.squarecamera.CameraFragment$4.onClick (CameraFragment.java:171)
android.view.View.performClick (View.java:6266)
android.view.View$PerformClick.run (View.java:24730)
android.os.Handler.handleCallback (Handler.java:789)
android.os.Handler.dispatchMessage (Handler.java:98)
android.os.Looper.loop (Looper.java:171)
android.app.ActivityThread.main (ActivityThread.java:6699)
java.lang.reflect.Method.invoke (Method.java)
com.android.internal.os.Zygote$MethodAndArgsCaller.run (Zygote.java:246)
com.android.internal.os.ZygoteInit.main (ZygoteInit.java:783)

and then this one

android.hardware.Camera.native_autoFocus (Camera.java)
android.hardware.Camera.autoFocus (Camera.java:1314)
com.desmond.squarecamera.SquareCameraPreview.handleFocus (SquareCameraPreview.java:182)
com.desmond.squarecamera.SquareCameraPreview.onTouchEvent (SquareCameraPreview.java:139)
android.view.View.dispatchTouchEvent (View.java:9301)
android.view.ViewGroup.dispatchTransformedTouchEvent (ViewGroup.java:2553)
android.view.ViewGroup.dispatchTouchEvent (ViewGroup.java:2254)
android.view.ViewGroup.dispatchTransformedTouchEvent (ViewGroup.java:2553)
android.view.ViewGroup.dispatchTouchEvent (ViewGroup.java:2254)
android.view.ViewGroup.dispatchTransformedTouchEvent (ViewGroup.java:2553)
android.view.ViewGroup.dispatchTouchEvent (ViewGroup.java:2254)
android.view.ViewGroup.dispatchTransformedTouchEvent (ViewGroup.java:2553)
android.view.ViewGroup.dispatchTouchEvent (ViewGroup.java:2254)
android.view.ViewGroup.dispatchTransformedTouchEvent (ViewGroup.java:2553)
android.view.ViewGroup.dispatchTouchEvent (ViewGroup.java:2254)
android.view.ViewGroup.dispatchTransformedTouchEvent (ViewGroup.java:2553)
android.view.ViewGroup.dispatchTouchEvent (ViewGroup.java:2254)
android.view.ViewGroup.dispatchTransformedTouchEvent (ViewGroup.java:2553)
android.view.ViewGroup.dispatchTouchEvent (ViewGroup.java:2254)
com.android.internal.policy.PhoneWindow$DecorView.superDispatchTouchEvent (PhoneWindow.java:2659)
com.android.internal.policy.PhoneWindow.superDispatchTouchEvent (PhoneWindow.java:1787)
android.app.Activity.dispatchTouchEvent (Activity.java:2771)
android.support.v7.view.WindowCallbackWrapper.dispatchTouchEvent (WindowCallbackWrapper.java:68)
com.android.internal.policy.PhoneWindow$DecorView.dispatchTouchEvent (PhoneWindow.java:2620)
android.view.View.dispatchPointerEvent (View.java:9521)
android.view.ViewRootImpl$ViewPostImeInputStage.processPointerEvent (ViewRootImpl.java:4238)
android.view.ViewRootImpl$ViewPostImeInputStage.onProcess (ViewRootImpl.java:4104)
android.view.ViewRootImpl$InputStage.deliver (ViewRootImpl.java:3650)
android.view.ViewRootImpl$InputStage.onDeliverToNext (ViewRootImpl.java:3703)
android.view.ViewRootImpl$InputStage.forward (ViewRootImpl.java:3669)
android.view.ViewRootImpl$AsyncInputStage.forward (ViewRootImpl.java:3795)
android.view.ViewRootImpl$InputStage.apply (ViewRootImpl.java:3677)
android.view.ViewRootImpl$AsyncInputStage.apply (ViewRootImpl.java:3852)
android.view.ViewRootImpl$InputStage.deliver (ViewRootImpl.java:3650)
android.view.ViewRootImpl$InputStage.onDeliverToNext (ViewRootImpl.java:3703)
android.view.ViewRootImpl$InputStage.forward (ViewRootImpl.java:3669)
android.view.ViewRootImpl$InputStage.apply (ViewRootImpl.java:3677)
android.view.ViewRootImpl$InputStage.deliver (ViewRootImpl.java:3650)
android.view.ViewRootImpl.deliverInputEvent (ViewRootImpl.java:5932)
android.view.ViewRootImpl.doProcessInputEvents (ViewRootImpl.java:5906)
android.view.ViewRootImpl.enqueueInputEvent (ViewRootImpl.java:5867)
android.view.ViewRootImpl$WindowInputEventReceiver.onInputEvent (ViewRootImpl.java:6035)
android.view.InputEventReceiver.dispatchInputEvent (InputEventReceiver.java:185)
android.os.MessageQueue.nativePollOnce (MessageQueue.java)
android.os.MessageQueue.next (MessageQueue.java:323)
android.os.Looper.loop (Looper.java:135)
android.app.ActivityThread.main (ActivityThread.java:5461)
java.lang.reflect.Method.invoke (Method.java)
com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:726)
com.android.internal.os.ZygoteInit.main (ZygoteInit.java:616)

can you help me how to fixed it?
thanks
